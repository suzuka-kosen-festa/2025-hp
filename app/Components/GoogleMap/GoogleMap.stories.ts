import type { Meta, StoryObj } from "@storybook/react";

import GoogleMap from "./GoogleMap";

type T = typeof GoogleMap;

const meta: Meta = {
  title: "Components/GoogleMap",
  component: GoogleMap,
  parameters: {
    screenshot: {
      skip: true,
    },
  },

  tags: ["autodocs"],
  argTypes: {
    address: {
      control: { type: "text" },
      description: "表示する住所",
    },
    width: {
      control: { type: "text" },
      description: "幅 (px, %, など)",
    },
    height: {
      control: { type: "number" },
      description: "高さ(px)",
    },
  },
} satisfies Meta<T>;

type Story = StoryObj<T>;

export const Default: Story = {
  args: {
    address: "三重県鈴鹿市白子町高専1-1 鈴鹿工業高等専門学校",
    height: 360,
  },
};

export const Tall: Story = {
  args: {
    address: "東京駅",
    height: 560,
  },
};

export const CustomWidth: Story = {
  args: {
    address: "京都駅",
    width: 600,
    height: 360,
  },
};

export default meta;
