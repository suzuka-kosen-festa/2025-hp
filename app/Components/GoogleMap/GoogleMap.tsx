import type { ReactNode } from "react";
import { Box } from "@mui/material";

interface GoogleMapProps {
  address: string
  width?: string | number
  height?: string | number
}

function GoogleMap({ address, width = "100%", height = 360 }: GoogleMapProps): ReactNode {
  const query = encodeURIComponent(address);
  const src = `https://www.google.com/maps?q=${query}&output=embed`;

  // レスポンシブな高さを設定
  const responsiveHeight = typeof height === "number"
    ? { xs: Math.min(height * 0.7, 250), sm: Math.min(height * 0.85, 320), md: height }
    : height;

  return (
    <Box sx={{
      width,
      height: responsiveHeight,
      borderRadius: { xs: "8px", sm: "12px" },
      overflow: "hidden",
      boxShadow: "0 4px 8px rgba(0,0,0,0.1)",
    }}
    >
      <iframe
        title="Google Map"
        src={src}
        width="100%"
        height="100%"
        style={{ border: 0 }}
        loading="lazy"
        referrerPolicy="no-referrer-when-downgrade"
        allowFullScreen
      />
    </Box>
  );
}

export default GoogleMap;
