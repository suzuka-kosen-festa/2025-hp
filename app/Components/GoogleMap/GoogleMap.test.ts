// @vitest-environment jsdom

import { composeStories } from "@storybook/react";
import { expect, it } from "vitest";

import * as stories from "./GoogleMap.stories";

const composed = composeStories(stories);

Object.entries(composed).forEach(([storyName, Story]) => {
  it(`googlemap snapshot: ${storyName}`, async () => {
    await Story.run();
    expect(document.body.firstChild).toMatchSnapshot();
  });
});
