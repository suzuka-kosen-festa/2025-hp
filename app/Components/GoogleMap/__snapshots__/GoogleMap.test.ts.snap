// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`googlemap snapshot: CustomWidth 1`] = `
<div>
  <div
    class="MuiBox-root css-b0f8qs"
  >
    <iframe
      allowfullscreen=""
      height="100%"
      loading="lazy"
      referrerpolicy="no-referrer-when-downgrade"
      src="https://www.google.com/maps?q=%E4%BA%AC%E9%83%BD%E9%A7%85&output=embed"
      style="border: 0px;"
      title="Google Map"
      width="100%"
    />
  </div>
</div>
`;

exports[`googlemap snapshot: Default 1`] = `
<div>
  <div
    class="MuiBox-root css-1vu43ej"
  >
    <iframe
      allowfullscreen=""
      height="100%"
      loading="lazy"
      referrerpolicy="no-referrer-when-downgrade"
      src="https://www.google.com/maps?q=%E4%B8%89%E9%87%8D%E7%9C%8C%E9%88%B4%E9%B9%BF%E5%B8%82%E7%99%BD%E5%AD%90%E7%94%BA%E9%AB%98%E5%B0%821-1%20%E9%88%B4%E9%B9%BF%E5%B7%A5%E6%A5%AD%E9%AB%98%E7%AD%89%E5%B0%82%E9%96%80%E5%AD%A6%E6%A0%A1&output=embed"
      style="border: 0px;"
      title="Google Map"
      width="100%"
    />
  </div>
</div>
`;

exports[`googlemap snapshot: Tall 1`] = `
<div>
  <div
    class="MuiBox-root css-13y2zt1"
  >
    <iframe
      allowfullscreen=""
      height="100%"
      loading="lazy"
      referrerpolicy="no-referrer-when-downgrade"
      src="https://www.google.com/maps?q=%E6%9D%B1%E4%BA%AC%E9%A7%85&output=embed"
      style="border: 0px;"
      title="Google Map"
      width="100%"
    />
  </div>
</div>
`;
